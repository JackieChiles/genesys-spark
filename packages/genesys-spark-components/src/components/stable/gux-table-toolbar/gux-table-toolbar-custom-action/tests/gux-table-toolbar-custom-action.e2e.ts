import { a11yCheck, newSparkE2EPage } from '../../../../../test/e2eTestUtils';

describe('gux-table-toolbar-custom-action', () => {
  const toolbarCustomAction =
    '<gux-table-toolbar-custom-action><span slot="text">Filter</span><gux-icon slot="icon" icon-name="filter" decorative></gux-icon></gux-table-toolbar-custom-action>';
  const toolbarCustomActionDisabled =
    '<gux-table-toolbar-custom-action disabled ><span slot="text">Filter</span><gux-icon slot="icon" icon-name="filter" decorative></gux-icon></gux-table-toolbar-custom-action>';

  describe('#render', () => {
    [
      {
        description: 'should render a toolbar-custom-action',
        html: toolbarCustomAction
      },
      {
        description: 'should render disabled toolbar-custom-action',
        html: toolbarCustomActionDisabled
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-table-toolbar-custom-action');

        await a11yCheck(page);
        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });

  describe('click', () => {
    it('should fire a click event when the toolbar action is clicked', async () => {
      const html = toolbarCustomAction;
      const page = await newSparkE2EPage({ html });
      const element = await page.find('gux-table-toolbar-custom-action');
      const onClickSpy = await element.spyOnEvent('click');

      await element.click();
      await page.waitForChanges();

      expect(onClickSpy).toHaveReceivedEventTimes(1);
    });

    it('should not fire a click event when the toolbar action is clicked', async () => {
      const html = toolbarCustomActionDisabled;
      const page = await newSparkE2EPage({ html });
      const element = await page.find('gux-table-toolbar-custom-action');
      const onClickSpy = await element.spyOnEvent('click');

      await element.click();
      await page.waitForChanges();

      expect(onClickSpy).toHaveReceivedEventTimes(0);
    });
  });
});
