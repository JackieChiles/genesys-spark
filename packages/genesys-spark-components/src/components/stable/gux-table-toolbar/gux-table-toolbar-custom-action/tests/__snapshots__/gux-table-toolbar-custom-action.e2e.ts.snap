// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-table-toolbar-custom-action #render should render a toolbar-custom-action 1`] = `"<gux-table-toolbar-custom-action hydrated=""><template shadowrootmode="open"><gux-button-slot accent="secondary" hydrated=""><button type="button" class="gux-action-title"><slot name="icon"></slot><span><slot name="text"></slot></span></button></gux-button-slot></template><span slot="text">Filter</span><gux-icon slot="icon" icon-name="filter" decorative="" size="inherit" hydrated=""></gux-icon></gux-table-toolbar-custom-action>"`;

exports[`gux-table-toolbar-custom-action #render should render disabled toolbar-custom-action 1`] = `"<gux-table-toolbar-custom-action disabled="" hydrated=""><template shadowrootmode="open"><gux-button-slot accent="secondary" hydrated=""><button disabled="" type="button" class="gux-action-title"><slot name="icon"></slot><span><slot name="text"></slot></span></button></gux-button-slot></template><span slot="text">Filter</span><gux-icon slot="icon" icon-name="filter" decorative="" size="inherit" hydrated=""></gux-icon></gux-table-toolbar-custom-action>"`;
